keras_nlp/preprocessing/lm_mask.py

```diff
@@ -0,0 +1,135 @@
+# Copyright 2022 The KerasNLP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Iterable
+
+import tensorflow as tf
+import tensorflow_text as tf_text
+from absl import logging
+from tensorflow import keras
+
+
+class LMMask(keras.layers.Layer):
+    """Class that applies language model masking.
+
+    This class is useful for preparing inputs for masked languaged modeling
+    (MLM) tasks. It follows the masking strategy described in the [original BERT
+    paper](https://arxiv.org/abs/1810.04805). Basically, given a tokenized text,
+    it randomly selects certain number of tokens for masking. Then for each
+    selected token, it has chance (configurable) to be replaced by "mask token"
+    or random token, or stay unchanged.
```

Comment:
What do the outputs of this layer look like? Do we want to stick with what tf text provides here?

Reply:
Yes, it will have the same outputs. I will update the docstring once we finalize our api interface design.